src/vulkan/resource.h

```diff
@@ -68,14 +69,7 @@ class Resource {
 
   Result MapMemory(VkDeviceMemory memory);
   void UnMapMemory(VkDeviceMemory memory);
-
-  // Set |memory_ptr_| as |ptr|. This must be used for only push constant.
-  // For Vulkan buffer and image i.e., Buffer and Image classes, we should
-  // not call this but uses MapMemory() method.
   void SetMemoryPtr(void* ptr) { memory_ptr_ = ptr; }
-
-  // Make all memory operations before calling this method effective i.e.,
-  // prevent hazards caused by out-of-order execution.
   void MemoryBarrier(CommandBuffer* command);
 
   uint32_t ChooseMemory(uint32_t memory_type_bits,
```

Comment:
This needs a comment.

> // Returns a memory index for the given Vulkan device, for a memory type satisfying the given constraints.

What the heck is force_flags for???  

If you don't know. add a TODO to document this function.

Reply:
Renamed the flag to require_flags_found and updated comment.